app/api/search/documentQueryBuilder.js

```diff
@@ -234,13 +250,24 @@ export default function() {
     },
 
     customFilters(filters = {}) {
-      Object.keys(filters).forEach(key => {
-        if (filters[key].values.length) {
-          addFilter({
-            terms: { [key]: filters[key].values },
-          });
-        }
-      });
+      Object.keys(filters)
+        .filter(key => filters[key].values.length)
+        .forEach(key => {
+          if (key === 'permissions.level') {
+            addFilter(
+              nested(
+                [
+                  { terms: { 'permissions.refId': permissionsContext.permissionsRefIds() } },
+                  { terms: { 'permissions.level': filters[key].values } },
+                ],
+                'permissions'
+              )
+            );
+            return;
```

Comment:
@mercy I believe there is a misconception of what a `return` will do inside a `forEach` loop here.  This return will skip the `addFilter` from line 269 when key is `permissions.level`, true... BUT... it will just skip for that element of the forEach, returning with the second, third, etc. elements afterwards... so, if key is, for example, `generatedToc` it will still work if it comes before or after the `permissions.level` occurrence.

Maybe I'm misreading what is your concern?

Reply:
You're right @RafaPolit, I'm sorry, I was effectively thinking of the break of the loop, but certainly, it doesn't happen in a forEach, I had a short circuit with this 😖